File: java/src/main/java/mx/iteso/adapter/WordDocAdapter.java
package mx.iteso.adapter;

/**
 * Creado por Lucifer.
 */
public class WordDocAdapter implements IGoogleDoc {
    /**
    * Word document.
    */
    private IWordDocument wordDocument;

    /**
    * License.
    */
    private MSLicense ms = new MSLicense("MS License");

    /**
    * Format.
    */
    private Format format;

    /**
    * Font.
    */
    private Font font;

    /**
     * Style.
     */
    private String style;

    /**
     * @param wordDoc Word Document.
     */
    public WordDocAdapter(final IWordDocument wordDoc) {
        this.wordDocument = wordDoc;
        font = new Font("Arial");
        style = "bold";
        format = new Format(font, style);

    }

    /**
     * @return Font.
     */
    public final Font getFont() {
        return format.returnFont();
    }

    /**
     * @return Style.
     */
    public final Object getStyle() {
        return format.getStyle();
    }

    /**
     * @return Background.
     */
    public final BackgroundImage getBackground() {
        return new BackgroundImage(wordDocument.getBackground());
    }

    /**
     * @param sharingPermissions Sharing Permissions.
     */
    public final void setSharingPermissions(final int sharingPermissions) {
        wordDocument.setMSOfficeVersion(sharingPermissions);
        wordDocument.getLicense();
        wordDocument.restrictEditIfLicenseIsInvalid(ms);

    }
}

File: csharp/Adapter Practice/WordDocAdapter.cs
namespace adapter_practice
{
    public class WordDocAdapter : IGoogleDoc {

        private readonly WordDocument _word;

        private readonly MsLicense _ms = new MsLicense("MS License");

        private readonly Format _format;

        private readonly Font _font;

        private readonly string _style;

        public WordDocAdapter(WordDocument word) {
            _word = word;
            _font = new Font("Arial");
            _style = "bold";
            _format = new Format(_font, _style);
        }

        public Font GetFont() {
            return _font;
        }

        public object GetStyle() {
            return _style;
        }

        public object GetFormat() {
            return _format;
        }

        public BackgroundImage GetBackground() {
            return new BackgroundImage(_word.GetBackground());
        }

        public void SetSharingPermissions(int sharingPermissions) {
            _word.SetMsOfficeVersion(sharingPermissions);
            _word.GetLicense();
            _word.RestrictEditIfLicenseIsInvalid(_ms);
        }

    }
}